### DISCUSSION TITLE
심사 전 테스트 API 호출 시 GW.AUTHN 오류가 반환되어 문의드립니다.

### DISCUSSION THREAD

**Discussion by kswjjg** [commerce-api-naver/commerce-solutions#99] [묻고 답하기]
호출 API는 /v1/commerce-solutions/external-transactions ( 외부 개발사 자체 결제 내역 전송 )이며

응답 값은 Array ( [code] => GW.AUTHN [message] => 요청을 보낼 권한이 없습니다. [timestamp] => 2024-12-26T17:26:19.469+09:00 [traceId] => pr1-185309-69bxbz^1733910841346^137807721 )

이렇게 반환되었습니다. 요청값 중 잘못된 부분이 있을까요? 통일한 토큰으로 /v1/seller/channels 는 호출되는걸 확인하였습니다.

**Comment by commerce-api-naver** [comment]
@kswjjg 

안녕하세요 커머스API 권현철입니다.

말씀해주신 내용에 대해 확인중으로 확인완료 시점에 답변 드리겠습니다.

감사합니다.

**Comment by commerce-api-naver** [comment]
@kswjjg 

안녕하세요 커머스API 권현철입니다.

말씀해주신 내용 확인 간 시간이 지연되어 죄송합니다.

'외부 개발사 자체 결제API' 호출 시 사용된 토큰의 인증 토큰 발급 타입(`type`)이 `SELLER`로 설정된 토큰이라 해당 오류가 발생하였습니다.
'커머스솔루션' API그룹과 관련된 API 호출 시에는 인증 토큰 발급 타입을 `SELF`로 설정되어야 정상 호출이 가능합니다.

- https://github.com/commerce-api-naver/commerce-api/discussions/780#discussion-5304047

번거로우시겠지만 API 호출 간 인증 토큰 발급 타입을 `SELF`로 설정한 토큰으로 '외부 개발사 자체 결제 내역 전송API' 호출 가능하실까요?
혹, 변경했음에도 동일한 오류가 발샌한다면 API 호출 건의 Trace ID 공유해 주시면 빠른 확인 도와드리겠습니다.
**TraceID는 모든 API 호출 건의 HTTP 응답 헤더(Response header) 내 GNCP-GW-Trace-ID 항목으로 제공됩니다.**

감사합니다.
